mbrl/models/model_trainer.py

```diff
@@ -190,15 +218,25 @@ def evaluate(self, dataset: TransitionIterator) -> torch.Tensor:
 
         batch_scores_list = []
         for batch in dataset:
-            avg_batch_score = self.model.eval_score(batch)
-            batch_scores_list.append(avg_batch_score)
-        batch_scores = torch.cat(batch_scores_list, axis=batch_scores_list[0].ndim - 2)
+            batch_score_and_maybe_meta = self.model.eval_score(batch)
+            if isinstance(batch_score_and_maybe_meta, tuple):
+                batch_score = cast(torch.Tensor, batch_score_and_maybe_meta[0])
+                meta = cast(Dict, batch_score_and_maybe_meta[1])
+            else:
+                # TODO remove this "else" in v0.2.0
+                warnings.warn(_NO_META_WARNING_MSG)
+                batch_score = cast(torch.Tensor, batch_score_and_maybe_meta)
+                meta = None
+            batch_scores_list.append(batch_score)
+            if batch_callback:
+                batch_callback(batch_score.mean(), meta)
```

Comment:
At the moment there is actually no indicator passed to the callback whether it is called from `evaluate()` or from `train()`. This makes it impossible to log training and evaluation metrics separately.
The callback should get some flag whether it is called from `evaluate()` or from `train()`. Or there should be separate callbacks for `train()` and `evaluate()`, but having three callbacks passed to `ModelTrainer` would probably be just more confusing and I would expect that the callbacks for training and evaluation typically should be very similar.

Reply:
I actually had this extra arg in the last commit, but after playing a bit with PlaNet I ended up removing it and force pushing an amend without it; the rationale was that this signaling could be taken care by the meta data itself (and same reasoning as above for keeping the API cleaner). 

Here I'm less sure though, and I'm inclined to add it back. One benefit of including it is that it makes it very explicit, so it's easy to avoid some silent error, and less responsibility on users to implement the metadata correctly.

Reply:
I would prefer having this argument. I think in all cases it is relevant whether the logged metrics were generated on training or evaluation data. So without the argument, in every case the user would have to do this bit of extra work of adding a flag to the dict, which could be avoided.

Reply:
Agreed, I'll add it.

Reply:
Added, and also a unit test.